### PR TITLE
Fix/delete temporary directories

### LINKED ISSUE #286: Old files under CONFIG_TEMP_DIR are not being deleted
This is from a k8s deployment where CONFIG_TEMP_DIR is set to /borges-tmp

```
/borges-tmp/tmp/672086548 # du -sh *
30.4G	local_repos
18.3G	transactioner

/borges-tmp/tmp/672086548 # date
Mon May 21 15:56:11 UTC 2018

/borges-tmp/tmp/672086548 # ls -lart transactioner/ | head -3
total 0
drwx------    4 root     root            80 May 20 22:43 ..
drwxr-xr-x    2 root     root            60 May 21 01:04 1da177e4c3f41524e886b7f1b8a0c1fc7321cac2_1526864649453136100

/borges-tmp/tmp/672086548 # ls -lart local_repos/  | head -3
total 0
drwx------    4 root     root            80 May 20 22:43 ..
drwxr-xr-x    4 root     root           120 May 20 22:44 01637e45-f585-be4d-400b-153ae91b4671_1526856270199338539
```

### PR DESCRIPTION
**Pull request by jfontan**
Fixes #286

### CODE REVIEW

**Comment by jfontan** [comment]
I had to rewrite the last commit as travis was having weird errors:

 `Could not fetch .travis.yml from GitHub.`

File: archiver.go [hunk 1]
```diff
@@ -152,6 +152,7 @@ func (a *Archiver) do(ctx context.Context, logger log.Logger, j *Job) (err error
 
 	log.Debugf("remote repository cloned")
 	if err := a.doPush(ctx, logger, &now, j, r, endpoint, gr); err != nil {
+		_ = gr.Close()
```
Comment (mcarmonaa) [inline archiver.go]: is there no need to check the error even just for logging it?

File: archiver.go [hunk 2]
```diff
@@ -400,6 +399,12 @@ func (a *Archiver) pushChangesToRootedRepository(ctx context.Context, logger log
 		}).Debugf("copy siva file from FS")
 		return err
 	})
+
+	if err != nil {
+		_ = tx.Rollback()
```
Comment (mcarmonaa) [inline archiver.go]: the same

**Comment by jfontan** [comment]
Added logging to tmp files deletion errors.

**Pull request merged**